app/notifications/controllers.py

```diff
@@ -61,17 +61,21 @@ def send_notifications(user):
 ##
 @listens_for(Notes, 'after_insert')
 def new_note(mapper, connection, new_note):
+    # Finds usernames when they are indicated with the @ symbol, stores them in 'mentioned'
     user_regex = "(?<=^|(?<=[^a-zA-Z0-9-_\.]))@([A-Za-z]+[A-Za-z0-9-_]+)"
     mentioned = re.findall(user_regex, new_note.description)
+    action = Actions.query.filter_by(id= int(new_note.action)).first()
 
     for u in mentioned:
         user = Users.query.filter_by(id= u).first()
-
         if user is not None:
             connection.execute(notifications_table, 
                 user = u,
                 type = NotificationType.MentionedInNote,
-                destination = new_note.id
+                destination = new_note.id,
+                viewed = False,
+                message = create_message(NotificationType.MentionedInNote, action.title),
+                redirect = create_redirect_string(NotificationType.MentionedInNote, action.charge)
             )
```

Comment:
NotificationType was used throughout because type is a reserved word in python and was not passing to the function correctly.